docs/src/snap/howto/install/install-offline.md

```diff
@@ -0,0 +1,262 @@
+used by its features (network, dns, etc) as well as any images that are
+needed to run your workloads.
+
+The following options are presented in the order of
+increasing complexity of implementation.
+You may also find it helpful to combine these options for your scenario.
+
+If you already have a running cluster,
+you can list the images in use by running:
+
+```bash
+sudo k8s list-images
+```
+
+Please remember to keep track of the images used by your workloads as well.
+
+#### Images Option A: via an HTTP proxy
+
+In many cases, the nodes of the airgap deployment may not have direct access to
+upstream registries, but can reach them through the
+[use of an HTTP proxy][proxy].
+
+The configuration of the proxy is out of the scope of this documentation.
+
+#### Images Option B: private registry mirror
+
+In case regulations and/or network constraints do not allow the cluster nodes
+to access any upstream image registry,
+it is typical to deploy a private registry mirror.
+This is an image registry service that contains all the required OCI Images
+(e.g. [registry](https://distribution.github.io/distribution/),
+[Harbor](https://goharbor.io/) or any other OCI registry) and
+is reachable from all cluster nodes.
+
+This requires three steps:
+
+1. Deploy and secure the registry service. This is out of the scope for this
+   document, please follow the instructions for the registry
+   that you want to deploy.
+2. Using [regsync][regsync], load all images from the upstream source and
+   push to your registry mirror.
+3. Configure the Canonical Kubernetes container runtime (`containerd`) to load 
+   images from
+   the private registry mirror instead of the upstream source. This will be
+   described in the installation section.
+
+In order to load images into the private registry, you need a machine with
+access to both the upstream registry (e.g. `docker.io`) and the internal one. 
+
+##### Load images with regsync
+
+We recommend using [regsync][regsync] to copy images
+from the upstream registry to your private registry.
+
+For the images used in the k8s-snap we currently sync upstream images
+to `ghcr.io/canonical`.
+
+Since you will need to do something similar you
+will find it helpful to look at the [upstream-images.yaml][upstream-imgs] file
+as well as the [sync-images][sync-images] script.
+
+In [upstream-images.yaml][upstream-imgs] you will have to
+change the sync target from `ghcr.io/canonical` to your private registry mirror.
+
+After you have updated the yaml file, you can run the [sync-images][sync-images]
+script:
+
+```bash
+USERNAME="$username" PASSWORD="$password" ./sync-images.sh
+```
+
+#### Images Option C: Side-load images
+
+Image side-loading is the process of loading all required OCI images directly
+into the container runtime, so that they do not have to be fetched at runtime.
+
+To create a bundle of images, you can use the [regctl][regctl] tool.
+
+```bash
+regctl image export --platform=local
```

Comment:
the correct way to invoke this would be through our tools, or point people to the upstream instructions instead.

```bash
./src/k8s/tools/regctl.sh ...
```

Two more notes for this section:
- The supplied command does not work, as it needs an image to sync (e.g. `ghcr.io/canonical/k8s-snap/pause:3.10`) as well as a target path.
- This section should not explain what side-loading is. Instead, it should document that:

> An alternative to configuring a registry mirror is to download all necessary OCI images, and then manually add them to all cluster nodes. Instructions for this are described in `[link to section]`.